src/main/java/picard/sam/SamAlignmentMerger.java

```diff
@@ -158,6 +141,70 @@ public void mergeAlignment(final File referenceFasta) {
         }
     }
 
+    @Override
+    protected SAMSequenceDictionary getDictionaryForMergedBam() {
+        SAMSequenceDictionary finalDict = null;
+        if (alignedSamFile != null && !alignedSamFile.isEmpty()) {
+            for (final File alignedFile : alignedSamFile) {
+                final SAMSequenceDictionary dict = SAMSequenceDictionaryExtractor.extractDictionary(alignedFile);
+                if (finalDict == null) {
+                    finalDict = dict;
+                } else {
+                    if (!compareSequenceDictionaries(finalDict, dict)) {
+                        log.error("Dictionaries of input files differ. Unable to continue.");
+                        throw new UnsupportedOperationException("Cannot use input files with different dictionaries.");
+                    }
+                }
+            }
+        } else {
+            final SeparateEndAlignmentIterator mergingIterator = new SeparateEndAlignmentIterator(this.read1AlignedSamFile, this.read2AlignedSamFile, referenceFasta);
+            finalDict = mergingIterator.getHeader().getSequenceDictionary();
+        }
+
+        SAMSequenceDictionary referenceDict = SAMSequenceDictionaryExtractor.extractDictionary(referenceFasta);
```

Comment:
Not sure I understand this comment. I cannot compare the SAM file dictionaries to the reference one since they might be different, but I do require that all the SAM file dictionaries are the same, and then I merge one of them with the reference dict....I could reverse the order (check that the reference dictionary exisits before checking that the sam files are the same), if that's what you mean.